clients/web/src/lib/span/normalize-spans.ts

```diff
@@ -23,32 +23,42 @@ export function computeColorClassName(
 }
 
 export function computeWaterfallStyle(
-  span: UiSpan,
+  span: Span,
   start: number,
   end: number,
   shortest?: number,
   longest?: number,
 ) {
   const callsContext = useCalls();
+  const {
+    monitorData: { durations },
+  } = useMonitor();
 
-  const offset = scaleNumbers([span.original.createdAt], start, end)[0];
+  const offset = scaleNumbers([span.createdAt], start, end)[0];
   const totalDuration = end - start;
 
-  const shortestSpanTime =
-    shortest ?? callsContext.durations.durations.shortestTime;
+  const shortestSpanTime = shortest ?? durations.shortestTime;
 
-  const longestSpanTime =
-    longest ?? callsContext.durations.durations.longestTime;
+  const longestSpanTime = longest ?? durations.longestTime;
 
   const granularity =
     ((longestSpanTime ?? 1) / (shortestSpanTime ?? 1)) *
     (callsContext.granularity.granularity() / 10000);
 
-  const width = Math.min(
-    scaleToMax([span.duration], totalDuration / granularity)[0],
-    100 - offset,
+  const scaledDuration =
+    callsContext.granularity.granularity() === 1
+      ? totalDuration
+      : totalDuration / granularity;
+
+  const spanDuration =
+    span.duration === -1 ? Date.now() * 1e6 - span.createdAt : span.duration;
```

Comment:
Why multiply by 1e6? Seems totally random. Same for line 81.

Reply:
Converting the timestamp to nanoseconds I think... This is something I kept from the early phases. Was not high on my priority list to investigate.